crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -111,6 +132,43 @@ struct TypeInferenceBuilder<'db> {
     types: TypeInference<'db>,
 }
 
+/// A builder is used by creating it with `::new`, and then calling `.finish()` on it, which
+/// returns the resulting [`TypeInference`].
+///
+/// There are a few different kinds of methods in the type inference builder, and the naming
+/// distinctions are a bit subtle.
+///
+/// The `finish` method calls `infer_region`, which delegates to one of `infer_region_scope`,
```

Comment:
```suggestion
/// The `finish` method calls [`infer_region`](TypeInferenceBuilder::infer_region), which delegates to one of `infer_region_scope`,
```

Or is this not possible because the method is `pub(crate)`?

Reply:
Seems to work fine. I learned a bit about rustdoc and discovered that in order to get it to actually render these docs I need `cargo doc -p red_knot_python_semantic --document-private-items` (the latter flag needed because this entire `infer` module is private).

This means that CI won't catch any warnings in these docs, because it doesn't build with `--document-private-items`. If we are writing these docs mostly for our own maintenance reasons anyway, maybe we should change that?

Reply:
I opened https://github.com/astral-sh/ruff/issues/12372 because it feels odd to be caring about rustdoc syntax in doc comments that our CI currently doesn't validate at all.